tests/modules/test_time.py

```diff
@@ -34,3 +31,147 @@ def test_sleep_negative(self):
                 print(err)
             print('Done.')
             """)
+
+    @unittest.expectedFailure
+    def test_struct_time_valid(self):
+        """
+        valid construction
+        """
+
+        # TODO: this won't pass until the follownig types have support __getitem__()
+        #     bytearray,
+        #     bytes,
+        #     dict,
+        #     frozenset,
+        #     range,
+        #     set,
+
+        seed = list(range(1, 10))
+        sequences = (
+            bytearray(seed),
+            bytes(seed),
+            dict(zip(seed, seed)),
+            frozenset(seed),
+            seed,
+            range(1, 10),
+            set(seed),
+            ''.join([str(i) for i in seed]),
+            tuple(seed)
+        )
+
+        for seq in sequences:
+            self.assertCodeExecution(struct_time_setup(seq))
+
+    def test_struct_time_valid_lengths(self):
+        """
+        length 9, 10, 11 are acceptable
+        """
+
+        for size in range(9, 12):
+            test_str = struct_time_setup([1] * size)
+            self.assertCodeExecution(test_str)
+
+    def test_struct_time_invalid(self):
+        """
+        invalid construction due to bad type
+        """
+
+        bad_types = [
+            True,
+            0j,
+            1,
+            None,
+            NotImplemented
+        ]
+
+        for t in bad_types:
+            self.assertCodeExecution(struct_time_setup(t))
```

Comment:
In terms of debugging - this will be easier to debug if you compile a *single* string, and run one `assertCodeExecution`, rather than iterate over a loop. In the loop case, the error will always occur on line 88, and it won't necessarily be clear *which* of the types is the problem. If it's a single call rolled out internally, you'll be able to see exactly which type is the problem.

As an added bonus - starting one VM and running lots of commands is faster than starting lots of individual VMs.